### PR TITLE
fix bugs with dubbo

### PR DESCRIPTION
**Pull request by yingziisme**
1. fix NullPointerException with dubbo getMetadata request

2. fix dubbo trace passing problem

the trace is third-demo -> consumer-demo -> provider-deme, but the parent of provider-demo is third-demo.
wrong data like this
image

after fixed 
image

### CODE REVIEW

**Comment by mateuszrzeszutek** [comment]
Hey @yingziisme can you sign the CLA?

**Comment by yingziisme** [comment]
> Hey @yingziisme can you sign the CLA?

ok. signed.

**Comment by mateuszrzeszutek** [comment]
Please run `./gradlew spotlessApply` on you PR, looks like the build is failing on broken formatting.

**Comment by yingziisme** [comment]
> Please run `./gradlew spotlessApply` on you PR, looks like the build is failing on broken formatting.
> 
> Also, can you add a regression test for your case? Especially the broken propagation one

Hey, Is it necessary to add the test? and do you mean the test with second bug?

**Comment by mateuszrzeszutek** [comment]
> Hey, Is it necessary to add the test? and do you mean the test with second bug?

It would be great if you could add one, cause it'd prevent this from happening again in case somebody modifies this instrumentation in the future.
Yeah, I meant the `dubbo trace passing problem` you mentioned in the OP

**Comment by yingziisme** [comment]
> > Hey, Is it necessary to add the test? and do you mean the test with second bug?
> 
> It would be great if you could add one, cause it'd prevent this from happening again in case somebody modifies this instrumentation in the future. Yeah, I meant the `dubbo trace passing problem` you mentioned in the OP

Hey, test has added

File: instrumentation/apache-dubbo-2.7/testing/src/main/groovy/io/opentelemetry/instrumentation/apachedubbo/v2_7/AbstractDubboTraceChainTest.groovy [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      .reference(middleReference)
+      .protocol(consumerProtocolConfig)
+      .start()
+
+    when:
+    GenericService genericService = middleReference.get()
+    def o = new Object[1]
+    o[0] = "hello"
+    def response = runWithSpan("parent") {
+      genericService.$invoke("hello", [String.getName()] as String[], o)
```
Comment (mateuszrzeszutek) [inline instrumentation/apache-dubbo-2.7/testing/src/main/groovy/io/opentelemetry/instrumentation/apachedubbo/v2_7/AbstractDubboTraceChainTest.groovy:112]: Nit: could probably be just 
```suggestion
    def response = runWithSpan("parent") {
      genericService.$invoke("hello", [String.getName()] as String[], ["hello"] as Object[])
```

File: instrumentation/apache-dubbo-2.7/testing/src/main/groovy/io/opentelemetry/instrumentation/apachedubbo/v2_7/AbstractDubboTraceChainTest.groovy [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    ServiceConfig<MiddleServiceImpl> service = new ServiceConfig<>()
+    service.setInterface(MiddleService)
+    service.setRef(new MiddleServiceImpl(genericService))
+    service.setRegistry(registerConfig)
+    return service
+  }
+
+  def "test apache dubbo base #dubbo"() {
+    setup:
+    def port = PortUtils.findOpenPort()
```
Comment (mateuszrzeszutek) [inline instrumentation/apache-dubbo-2.7/testing/src/main/groovy/io/opentelemetry/instrumentation/apachedubbo/v2_7/AbstractDubboTraceChainTest.groovy:74]: Nit: you could find both open ports at the same time:
```suggestion
    def port = PortUtils.findOpenPorts(2)
    def middlePort = port + 1
```

File: instrumentation/apache-dubbo-2.7/testing/src/main/groovy/io/opentelemetry/instrumentation/apachedubbo/v2_7/AbstractDubboTraceChainTest.groovy [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    def registerConfig = new RegistryConfig()
+    registerConfig.setAddress("N/A")
+    ServiceConfig<MiddleServiceImpl> service = new ServiceConfig<>()
+    service.setInterface(MiddleService)
+    service.setRef(new MiddleServiceImpl(genericService))
+    service.setRegistry(registerConfig)
+    return service
+  }
+
+  def "test apache dubbo base #dubbo"() {
```
Comment (mateuszrzeszutek) [inline instrumentation/apache-dubbo-2.7/testing/src/main/groovy/io/opentelemetry/instrumentation/apachedubbo/v2_7/AbstractDubboTraceChainTest.groovy:72]:
```suggestion
  def "test that context is propagated correctly in chained dubbo calls"() {
```

**Review by mateuszrzeszutek — APPROVED** [review]
Looks great, thanks!
Left a few minor comments, but overall it LGTM 👍

**Comment by yingziisme** [comment]
> Looks great, thanks! Left a few minor comments, but overall it LGTM 👍

ok. these conversations have resolved.

**Review by mateuszrzeszutek — APPROVED** [review]
Thanks!

**Pull request merged**